### PR TITLE
blindtree merkle fork changes

### PR DESCRIPTION
**Pull request by heronhaye**

### CODE REVIEW

File: hash.go [hunk 1]
```diff
@@ -7,15 +7,12 @@ import (
 // Len returns the number of bytes in the hash, but after shifting off
 // leading 0s from the length size of the hash
 func (h Hash) Len() int {
```
Comment (heronhaye) [inline hash.go]: was taking up a ton of cpu time the old way

Comment (heronhaye) [reply hash.go]: added a test too

**Comment by maxtaco** [comment]
is the plan to have 2 different package going forward that differ in their camel-casing?

**Comment by maxtaco** [comment]
... and why do we need a fork?

**Comment by heronhaye** [comment]
I originally had more significant changes but I managed to move them to the engine interface in kbweb instead. I can change back to go-merkle-tree and update the existing kbfs code to match the new types.

**Comment by maxtaco** [comment]
thanks, i'd prefer that....

**Pull request closed without merging**